crypto/dilithium/pqcrystals_dilithium_ref_common/rounding.c

```diff
@@ -76,27 +80,31 @@ unsigned int make_hint(int32_t a0, int32_t a1) {
 *
 * Description: Correct high bits according to hint.
 *
-* Arguments:   - int32_t a: input element
+* Arguments:   - ml_dsa_params: parameter struct
+*              - int32_t a: input element
 *              - unsigned int hint: hint bit
 *
 * Returns corrected high bits.
 **************************************************/
-int32_t use_hint(int32_t a, unsigned int hint) {
+int32_t use_hint(ml_dsa_params *params, int32_t a, unsigned int hint) {
   int32_t a0, a1;
 
-  a1 = decompose(&a0, a);
+  assert((params->gamma2 == (Q-1)/32) || (params->gamma2 == (Q-1)/88));
+
+  a1 = decompose(params, &a0, a);
   if(hint == 0)
     return a1;
 
-#if GAMMA2 == (Q-1)/32
-  if(a0 > 0)
-    return (a1 + 1) & 15;
-  else
-    return (a1 - 1) & 15;
-#elif GAMMA2 == (Q-1)/88
-  if(a0 > 0)
-    return (a1 == 43) ?  0 : a1 + 1;
-  else
-    return (a1 ==  0) ? 43 : a1 - 1;
-#endif
+  if (params->gamma2 == (Q-1)/32) {
+    if(a0 > 0)
+      return (a1 + 1) & 15;
+    else
+      return (a1 - 1) & 15;
+  }
+  else  {
```

Comment:
This is a slight behavior difference, does anything enforce `GAMMA2` == `(Q-1)/32` or `(Q-1)/88`?

Reply:
Line 92 above asserts exactly that.

Reply:
We control parameters and run KATs. This should be quite safe to do.

Reply:
Agreed! We set `gamma2` ourselves, and I've added references to the standard to show where it comes from.